aspnetcore/blazor/fundamentals/dependency-injection.md

```diff
@@ -317,13 +317,15 @@ public IMyService MyService { get; set; }
 
-Even in client-side Blazor apps that don't operate over a circuit, services registered with a scoped lifetime are treated as singletons, so they live longer than scoped services in typical ASP.NET Core apps. Client-side transient services can also live longer than expected because there's no request-response-based lifetime to trigger DI system disposal of transient services. Although long-lived transient services are of greater concern on the server, they should generally be avoided as client service registrations as well. Use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type is also recommended for client-side services to control service lifetime.
+Even in client-side Blazor apps that don't operate over a circuit, services registered with a scoped lifetime are treated as singletons, so they live longer than scoped services in typical ASP.NET Core apps. Client-side disposable transient services can also live longer than expected because the DI container persists for the lifetime of the app. Although long-lived disposable transient services are of greater concern on the server, they should be avoided as client service registrations as well. Use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type is also recommended for client-side services to control service lifetime.
 
-An approach that limits a service lifetime is use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type. <xref:Microsoft.AspNetCore.Components.OwningComponentBase> is an abstract type derived from <xref:Microsoft.AspNetCore.Components.ComponentBase> that creates a DI scope corresponding to the *lifetime of the component*. Using this scope, it's possible to use DI services with a scoped lifetime and have them live as long as the component. When the component is destroyed, services from the component's scoped service provider are disposed as well. This can be useful for services that:
+An approach that limits a service lifetime is use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type. <xref:Microsoft.AspNetCore.Components.OwningComponentBase> is an abstract type derived from <xref:Microsoft.AspNetCore.Components.ComponentBase> that creates a DI scope corresponding to the *lifetime of the component*. Using this scope instead of a scoped service or disposable transient service, a component can inject services with a scoped lifetime and have them live as long as the component. When the component is destroyed, services from the component's scoped service provider are disposed as well. This can be useful for services that:
```

Comment:
> Using this scope instead of a scoped service or disposable transient service, a component can inject services with a scoped lifetime and have them live as long as the component.

=>

> Using this scope for a scoped service or disposable transient service, a component can inject services with a limited lifetime and have them live as long as the component.

Reply:
I think I like mine better. 😄 I don't want to imply that `ScopedService` should try to take a disposable transient service. Therefore, I like "instead of" ... ***however*** I do want to change it to use `OwningComponentBase` over "scope" ...

```suggestion
An approach that limits a service lifetime is use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type. <xref:Microsoft.AspNetCore.Components.OwningComponentBase> is an abstract type derived from <xref:Microsoft.AspNetCore.Components.ComponentBase> that creates a DI scope corresponding to the *lifetime of the component*. Using this scope instead of directly injecting a scoped service or using a disposable transient service, a component can inject services with a scoped lifetime and have them live as long as the component. When the component is destroyed, services from the component's scoped service provider are disposed as well. This can be useful for services that:
```

Reply:
It can't AFAIK ... AFAIK transient disposables shouldn't be used at all, and that's what the text says. However, @MackinnonBuck can set me straight on it.

Reply:
I did the test and `OwningComponentBase` disposes the transient disposable.
- [TransientDisposableService.cs](https://github.com/hakenr/BlazorTransientDisposablesTest/blob/master/BlazorTransientDisposables/BlazorTransientDisposables.Client/TransientDisposableService.cs)
- [Home.razor](https://github.com/hakenr/BlazorTransientDisposablesTest/blob/master/BlazorTransientDisposables/BlazorTransientDisposables.Client/Pages/Home.razor)
- [Program.cs](https://github.com/hakenr/BlazorTransientDisposablesTest/blob/master/BlazorTransientDisposables/BlazorTransientDisposables.Client/Program.cs)

![image](https://github.com/dotnet/AspNetCore.Docs/assets/12828888/c2fa9bec-cf06-446e-8b2a-c18ddd9c5d15)

So I recommend to express you can use it to "fix" the evil transient disposable (coming from code out of your control).
;-)

Reply:
I see .....

Mackinnon might be able to make a ruling on this ... Javier originally said ***Do NOT use them on PAIN of DEATH 💀!*** .... well 😆 ... I added the "pain of death" part 😆. If you're not sure Mackinnon, I have an offline email convo with Javier from earlier that I can ping him on to ask a follow-up question about this ... can `OwningComponentBase`/`ScopedServices` be safely used for disposable transients?